### ISSUE TITLE
Two dataloader for one dataset with different augmentation

### ISSUE THREAD

**Issue by shigen-StoneRoot** [MIC-DKFZ/nnUNet#1649]
Hi,
I want to obtain two dataloader for one dataset with different augmentation strategies, such as cropping size.  The challenge lies in the fact that I don’t know how to make two loader generate augmented data for the same subject. The  generate_train_batch() function in dataset_loading.py shows that the key is randomly selected. How to make two loaders generate augmented data for the same sample simultaneously?

Thanks!

**Comment by norajitr** [comment]
Hi shigen-StoneRoot,

for your issue, you can probably follow the documentation's advice and use your own trainer class:
https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/extending_nnunet.md

"- If you intend to modify the training procedure (loss, sampling, data augmentation, lr scheduler, etc) then you need 
to implement your own trainer class. Best practice is to create a class that inherits from nnUNetTrainer and 
implements the necessary changes. Head over to our [trainer classes folder](../nnunetv2/training/nnUNetTrainer) for 
inspiration! There will be similar trainers for what you intend to change and you can take them as a guide. nnUNetTrainer 
are structured similarly to PyTorch lightning trainers, this should also make things easier!"

In your own trainer class, you can also customize the type of augmentations you intend to apply.

You can get inspiration from the custom trainer classes in this folder: nnunetv2/training/nnUNetTrainer/variants/data_augmentation

I hope this helps.

Best,
Tobias
